### PR TITLE
Reporting task changes

### PR DESCRIPTION
**Pull request by poojabh91**

### CODE REVIEW

File: DCode.Web/Security/AuthorizeReportingTask.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        public override void OnAuthorization(AuthorizationContext filterContext)
+        {
+
+            var userIdentity = ConfigurationManager.AppSettings[Constants.UseWindowsIdentity].Equals(Constants.True) ? WindowsIdentity.GetCurrent() : HttpContext.Current.User.Identity;
+            if (userIdentity != null && userIdentity.IsAuthenticated)
+            {
+                var commonService = _kernel.Value.Get<ICommonService>();
+
+                //if (SessionHelper.Retrieve(Constants.UserContext) == null)
+                {
```
Comment (ritwiksen) [inline DCode.Web/Security/AuthorizeReportingTask.cs:47]: Please remove braces as `if` is commented out.

File: DCode.Web/Security/AuthorizeReportingTask.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+                    SessionHelper.Save(Constants.UserContext, userContext);
+                }
+            }
+            else
+            {
+                filterContext.Result = new HttpUnauthorizedResult();
+                return;
+            }
+
```
Comment (ritwiksen) [inline DCode.Web/Security/AuthorizeReportingTask.cs]: Please remove all empty lines.

File: DCode.Web/Views/Reporting/SplitPerOffering.cshtml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿@model IEnumerable<SearchApplication.Models.SplitPerOffering>
+
+@{
+    ViewBag.Title = ViewBag.PageTitle;
+    Layout = "~/Views/Shared/_LayoutReporting.cshtml";
+}
+
```
Comment (ritwiksen) [inline DCode.Web/Views/Reporting/SplitPerOffering.cshtml]: Please remove empty line breaks across the board.

File: DCode.Web/Views/Reporting/TopContributors.cshtml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            <th>
+                @Html.ActionLink("Export To Excel", "ExportToExcel", "Reporting")
+            </th>
+        </tr>
+    </table>
+
+</form>
+
+
+<table class="table table-bordered table-responsive table-hover">
```
Comment (ritwiksen) [inline DCode.Web/Views/Reporting/TopContributors.cshtml:8]: Can we componentize this table in a partial and use it in `TopContributors.cshtml` , `TopRequestors.cshtml` &  `SplitPerOffering.cshtml`.

File: DCode.Web/Views/ReportingBase/Base.cshtml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+</head>
+<body>
+    <div class="navbar navbar-inverse navbar-fixed-top">
+      Reporting content here   
+    </div>
+
+  
+
+
+    @Scripts.Render("~/bundles/jquery")
```
Comment (ritwiksen) [inline DCode.Web/Views/ReportingBase/Base.cshtml]: `jquery` & `bootstrap` have already been loaded in `_layoutReporting.cshtml`. Is there a specific need to re-render them?

File: SearchApplication/Controllers/ReportingController.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+namespace SearchApplication.Controllers
+{
+    public class ReportingController : Controller
+    {
+        
+       
+        // GET: Reporting
+        public ActionResult TopRequestors()
+        {
+            string conString = System.Configuration.ConfigurationManager.ConnectionStrings["DcodeConnectionString"].ToString();
```
Comment (ritwiksen) [inline SearchApplication/Controllers/ReportingController.cs]: A couple of observations here:

1) Can we use `using` blocks to auto-dispose of connection and command objects. 
2) Also, pulling up EDM objects directly into the controller layer isn't a good idea.
3) Would be best if we closed connections in a `finally` block.
4) Can store the connection string as a constant at the controller level.

File: SearchApplication/Controllers/TasksController.cs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            Entities db = new Entities();
+            IEnumerable<task_type> taskTypesList = db.task_type.ToList();
+            IEnumerable<offering> offeringsList = db.offerings.ToList();
+            //IEnumerable<service_line> serviceLineList = db.service_line.ToList();
+            ViewBag.taskTypesList = taskTypesList;
+            ViewBag.offeringsList = offeringsList;
+            //ViewBag.serviceLine = serviceLineList;
+
+            SearchModel searchModel = new SearchModel();
+            if (fm["ddl_Task_Type"].ToString() != "")
```
Comment (ritwiksen) [inline SearchApplication/Controllers/TasksController.cs]: Please use `Convert.ToString()` in place of `.ToString()`.

File: SearchApplication/Global.asax.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (ritwiksen) [inline SearchApplication/Global.asax.cs]: Please remove unused `usings`.

File: SearchApplication/Views/Reporting/SplitPerOffering.cshtml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            <th>
+                @Html.ActionLink("Export To Excel", "ExportToExcel", "Reporting")
+            </th>
+        </tr>
+    </table>
+
+</form>
+
+
+<table class="table table-bordered table-responsive table-hover">
```
Comment (ritwiksen) [inline SearchApplication/Views/Reporting/SplitPerOffering.cshtml:7]: Table can be reused across multiple views.

File: SearchApplication/Views/Security/AuthorizeReportingTask.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        {
+
+            var commonService = _kernel.Value.Get<ICommonService>();
+            // var userContext = _commonService.GetCurrentUserContext();
+            //var userName = userIdentity.Name.Substring(userIdentity.Name.IndexOf(@"\") + 1);
+            var userContext = commonService.GetCurrentUserContext();
+
+            var internalUsers = ConfigurationManager.AppSettings["ReportingUsers"];
+
+            if (internalUsers != null
```
Comment (ritwiksen) [inline SearchApplication/Views/Security/AuthorizeReportingTask.cs:44]: Considering that this split was used at multiple places, could we think of creating an extension method for this.

**Review by ritwiksen — CHANGES_REQUESTED** [review]
There are no. of changes that are needed mostly across the board. A major issue is that we are consuming EDM objects directly in the Controller layer.

**Review by shivam-r — APPROVED** [review]

**Pull request merged**